clang/lib/Sema/SemaChecking.cpp

```diff
@@ -8092,6 +8092,18 @@ void Sema::checkCall(NamedDecl *FDecl, const FunctionProtoType *Proto,
     diagnoseArgDependentDiagnoseIfAttrs(FD, ThisArg, Args, Loc);
 }
 
+void Sema::CheckConstrainedAuto(TypeSourceInfo *TS) {
+  if (getLangOpts().CPlusPlus20) {
+    if (const AutoType *AutoT = TS->getType()->getAs<AutoType>()) {
```

Comment:
I think this check should be kept on the call site and the function can take an `AutoType` as parameter

Reply:
Fixed.